js/packages/eyes-cypress/src/browser/mergeCypressConfigs.ts

```diff
@@ -0,0 +1,37 @@
+import type {CypressEyesConfig} from '../expose'
+type BatchProp = 'batchName' | 'batchId' | 'batchSequence' | 'batchSequenceName' | 'notifyOnCompletion'
+type BatchNestedProp = 'name' | 'id' | 'sequenceName' | 'notifyOnCompletion'
+
+export function mergeCypressConfigs({
+  globalConfig,
+  openConfig,
+}: {
+  globalConfig: CypressEyesConfig
+  openConfig: CypressEyesConfig
+}): CypressEyesConfig {
+  const batch = {
+    id: getBatchProperty('batchId', 'id'),
+    name: getBatchProperty('batchName', 'name'),
+    sequenceName: getBatchProperty('batchSequenceName', 'sequenceName'),
+    notifyOnCompletion: getBatchProperty('notifyOnCompletion', 'notifyOnCompletion'),
+  }
+
+  return {
+    ...globalConfig,
+    ...openConfig,
+    batch,
+  }
+
+  function getBatchProperty(prop: BatchProp, nestedProp: BatchNestedProp): any {
+    if (openConfig.hasOwnProperty(prop)) {
+      return openConfig[prop]
+    } else if (openConfig.batch && openConfig.batch.hasOwnProperty(nestedProp)) {
+      return openConfig.batch[nestedProp]
+    } else if (globalConfig.hasOwnProperty(prop)) {
+      return globalConfig[prop]
+    } else if (globalConfig.batch && globalConfig.batch.hasOwnProperty(nestedProp)) {
+      return globalConfig.batch[nestedProp]
+    }
+    return undefined
+  }
```

Comment:
This seems like an overcomplication to me. Also, you don't need to transform config here, you should simply merge it, which means you don't need to put `batchId` prop in the `batch` object, you will do it when you transform `CypressConfig` to `Config` (`CoreConfig`).

```suggestion
  return {
    ...globalConfig,
    ...openConfig,
    batch: {...globalConfig.batch, ...openConfig.batch}
  }
```

Reply:
@kyrylodolynskyi:
1) Doing what you suggested and doing `{...globalConfig, ....openConfig}` is the same.
2) The logic I presented here is merging , for instance, if the user set in the global config `batch: {id: '1234', name: 'my batch'}` and in open `batchId: '4321'`, I'll need to overwrite the id from the global with the prop I got from open and still keep the `name`

Reply:
1. It is not the same, because you will get the `batch` object as a composition of `batch` from `globalConfig` and `openConfig`.
2. You are not exactly merging here, you are transforming. The return value of this function is not `CypressConfig`, it is a different type that doesn't have `batchId`, `batchName`, `batchSequenceName`, `notifyOnCompletion` properties (it still has but we don't need to look at them) because they we already processed and used to construct `batch` object. I don't want to introduce a new type, I want us to use only what is necessary, which means in this case we should minimize the work with cypress specific type. If we going to transform this type into something else anyway, why would we split this work? All of the transformations should be done in the `transfromCypressConfig` function.

Reply:
@kyrylodolynskyi ok, I see your pooint